### ISSUE TITLE
Weapon Malfunction Target is Blocked error breaks turrets

### ISSUE THREAD

**Issue by IamGoot** [RavienGaming/Reforged-Eden-2#28]
### Reforged Eden 2 Build Number

Reforged Eden 2 Alpha (Build 1)

### Game mode

Survival, Single Player

### Tested in vanilla

Not tested in vanilla or not applicable

### Description of issue

I have posted this issue already but have no clue where Github put it. I have added some information in this post so its slightly different from the earlier post.

Getting a Weapon Malfunction notification from my SV Light Laser Turrets, with a notification of "line of sight blocked" ..... Upon receiving this, the weapon is useless, the AI won't fire at anything, on occasion wont track anything and they cannot be repaired, either with a Multitool (I am using Advanced) or at a T2 Repair bay.

They show as not needing repair

If I attempt to manually control them, they fire and rotate, but the AI of the game will not fire at anything and sometimes won't track targets. When testing manually, I found the turrets gave me the Weapon Malfunction notification, if I tried to fire downwards into the ship, but the targets when the AI is controlling the turrets are purposely being positioned above my SV where there should be a clear line of sight but they refuse to fire giving the Weapon Malfunction Notification. I have checked that they have ammo and the relevant targetting options are set correctly

Its like as though when the AI is using them, the AI thinks the weapon is looking into the ship when the graphics show its aiming at a target.

Something there must be wrong. Never had this notification in 3000+ hours of the game both Vanilla or RE1 prior to seeing it in this scenario

Things I've tried.

I have changed the location of my turrets to see if other blocks are somehow blocking line of sight, this turned out to not change anything

For reference the SV was made within a RE2 creative session and isn't a conversion from RE1 or vanilla.

Playing with CPU turned on, CPU limit of SV is under 14000 CPU (14000 current max)

### Error code from console or log (Optional)

No idea where to find this, as my logs are total gibberish to me.

**Comment by VermillionX** [comment]
Turret obstruction checks are all built into the model that eleon provides. We cannot change or alter the behavior of the turret AI or it's ability to track targets, making this a Game Bug.

**Comment by RavienGaming** [comment]
Also potentially related bug report:
https://empyriononline.com/threads/turret-line-of-sight-is-incorrect-while-seated-in-cockpit-00406.100745/

There was at least one other bug report about a different turret issue but I could not find it when searching.

**Comment by IamGoot** [comment]
I see a similarity in the link you've provided Ravien. However the turrets I am using won't fire automatically at any angle, whether the target is in front, side, rear, at an angle above or directly above but within the turrets rotational engagement range.

I have over 3000 hours in the game. I accept you will have 10 times this.... I have NEVER had this issue in any other version of the game. RE1, vanilla, any of the other scenarios I have played, none of them. I regularly build mining SVs with retractable turrets behind the cockpit, because thats streamline when digging deep holes so the ship doesn't snag on rock, with turrets being deployed if I become under attack. I have only ever received the weapon malfunction issue in this scenario.

**Comment by IamGoot** [comment]
Took my SV into a vanilla game, minor modifications were obviously needed, converting RE2 to vanilla.... turrets worked just fine, no issues, no weapon malfunction issues, no AI issues firing at targets automatically, no tracking issues, nothing.

As expected

**Comment by RavienGaming** [comment]
Yeah it's still a game bug because there's nothing that can be changed within a scenario. All we can do is set the model used for the turret, all the scripts used for the actual turret controller are not accessible to us.

**Comment by IamGoot** [comment]
It just leaves doubt in my mind as to how overall combat is going to pan out in the overall scenario. If the AI is going to say "nothing" is blocking its line of sight to a target, Its going to have a direct fire outcome to most fights. If thats SV restricted, it might be manageable, but if that b*llsh*t migrates to CV combat.... thats just going to ruin it for me, cos I aint manuevering a CV down the beasts throat to fire direct fire weapons, because the AI can't be assed

**Comment by RavienGaming** [comment]
Yeah unfortunately there's nothing we can do to fix it. It's all controlled by Eleon's models and scripts which we do not have access to. Nor can we make our own custom turrets. There have been several bug reports filed on various weapon (turret and fixed) problems so the best we can hope for is to file bug reports and hope they can fix the issues.

However if you find issues with custom fixed weapon models, those we may be able to fix if it's a problem with our own models. But anything that uses a vanilla model is hard coded, unchangeable, and we don't even have access to them in Unity, and would need to be reproduced and reported in vanilla at: https://empyriononline.com/forums/bugs.24/

**Comment by Ente2k** [comment]
Can't you guys not just try to use a different model for the laser Turret? The other turrets don't seem to have that issue. And there is really little hope that eleon will fix that specifc bug....
In any case it is weird that only the light laser turret seems to have that issue, when the medium turret uses the same model.

**Comment by Ente2k** [comment]
Ok Scratch that model swap, changing the model doesnt help.
But, you do have a suspicious 
`CheckObstructed: true`
in the itemconfig for the laser turrets weapon. If I do comment that line, I, at least for my one testflight, no longer had the issue.

**Comment by IamGoot** [comment]
> Ok Scratch that model swap, changing the model doesnt help. But, you do have a suspicious `CheckObstructed: true` in the itemconfig for the laser turrets weapon. If I do comment that line, I, at least for my one testflight, no longer had the issue.

Is that a command line I can manipulate or is it buried in modding code?

Sounds nice to read someone else has encountered the same problem, I was beginning to think I was going mad

**Comment by Ente2k** [comment]
Well its in the modding code, in the itemconfig.ecf. In the current one on or close to line 10062. You could modify it, but you would have to redo the modification each time you update the scenario. 
Mind also it might not even fix the issue, I need to do more tests to see whether it is really gone. 

Would need @VermillionX  to look into it whether that is indeed the culprit. Could very well be I am just an idiot agian.

**Comment by VermillionX** [comment]
It's a vanilla thing. I'm pretty sure if it's NOT there, you can stick your guns inside the hull. But who knows at this point.

**Comment by RavienGaming** [comment]
I'll look into it but it'll be a few days.

**Comment by VermillionX** [comment]
Removed from the turrets for the next update. May or may not fix the problem.

**Comment by Ente2k** [comment]
Well as long as you only remove it from the weapon for the turret it shouldn't allow fixed weapons to be sticked inside the hull. And the other turrets also don't have it, I think. But who knows.

**Comment by RavienGaming** [comment]
Right most turrets do not have that check, in both RE1/2 and vanilla, so we can assume it's safe to remove it. But whether it will fix all the issues with blocked firing directions is unknown. (Probably not).

**Comment by IamGoot** [comment]
I've pondered if its a "retractable turret" issue.... I predominantly use retractable turrets, as I've noticed this on my mining SV mostly, so don't norminally mount "sit on the hull" turrets

**Comment by RavienGaming** [comment]
Merging several other issues with "blocked firing" into this one to consolidate.

**Comment by Ente2k** [comment]
I havent seen this particular thing see happen with the sv light laser turret, since you removed the weapon blocked check. So unless someone tells me how wrong I am, I would say this report can be marked as fixed (and probably also have the game bug removed, since it was indeed the scenarios fault, in this particular case)

Also, in case anyone wonders:
#248 was already stated by the devs as working as intended. So not a bug (just shitty implementation in the game). Source: https://empyriononline.com/threads/hitscan-weapons-check-hits-against-the-ship-firing-them.99930/#post-456770
#250, #79, #70 are indeed all different base game issue that should be reported if someone manages to do a reproduction in vanilla.

**Comment by RavienGaming** [comment]
Thanks Ente. Was trying to cut down on the number of these related but weird open issues a bit.
